modules/dbparser/stateful-parser.c

```diff
@@ -0,0 +1,54 @@
+#include "stateful-parser.h"
+#include <string.h>
+
+
+void
+stateful_parser_set_inject_mode(StatefulParser *self, const gchar *inject_mode)
+{
+  self->inject_mode = stateful_parser_lookup_inject_mode(inject_mode);
```

Comment:
Is the `-1` value handled correctly? The default is `pass-through` so I'd expect to fallback to that. `stateful_parser_emit_synthetic()` will create an internal message in this case.